src/System.Text.Json/src/System/Text/Json/Serialization/Converters/DefaultImmutableConverter.cs

```diff
@@ -104,31 +134,24 @@ public override IEnumerable CreateFromList(ref ReadStack state, IList sourceList
             Type elementType = state.Current.GetElementType();
 
             string delegateKey = GetDelegateKey(immutableCollectionType, elementType, out _, out _);
-            Debug.Assert(CreateRangeDelegates.ContainsKey(delegateKey));
+            Debug.Assert(s_createRangeDelegates.ContainsKey(delegateKey));
 
             JsonClassInfo elementClassInfo = state.Current.JsonPropertyInfo.ElementClassInfo;
+            JsonPropertyInfo propertyInfo = DefaultIEnumerableConstructibleConverter.GetElementJsonPropertyInfo(elementClassInfo, options);
```

Comment:
This is an improvement over accessing the cache on that class directly 👍

Reply:
Whoop. If one more converter or some other type needs this, I'll extract it to it's own cache class.